### ISSUE TITLE
keras Sequential model get empty metrics_names

### ISSUE THREAD

**Issue by zishuaiz** [tensorflow/tensorflow#43142]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): macOS Catalina 0.15.5
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: NA
- TensorFlow installed from (source or binary): pip3 install --upgrade tensorflow
- TensorFlow version (use command below): 2.3.0
- Python version:
```
Python 3.7.5 (default, Oct 25 2019, 10:52:18)
[Clang 4.0.1 (tags/RELEASE_401/final)] :: Anaconda, Inc. on darwin
```
- Bazel version (if compiling from source): NA
- GCC/Compiler version (if compiling from source): NA
- CUDA/cuDNN version: NA
- GPU model and memory: NA

**Describe the current behavior**
```python
def build_model():
    model = keras.Sequential()
    model.add(keras.layers.Embedding(vocab_size, 64))
    model.add(keras.layers.GlobalAveragePooling1D())
    model.add(keras.layers.Dense(128, activation='relu'))
    model.add(keras.layers.Dense(128, activation='relu'))
    model.add(keras.layers.Dense(64, activation='relu'))
    model.add(keras.layers.Dense(1))
    optimizer = tf.keras.optimizers.RMSprop(0.01)
    model.compile(loss='mse',
                  optimizer=optimizer,
                  metrics=['mae', 'mse'])
    return model


model = build_model()
print(model.metrics_names)  #  print []
```

**Describe the expected behavior**
should print out: ['loss', 'mae', 'mse']
tensorflow 2.1.0 print correct

**Comment by amahendrakar** [comment]
@zishuaiz,
As per the [documentation](https://www.tensorflow.org/api_docs/python/tf/keras/Model#attributes_1), `metrics_names` are available only after a `keras.Model` has been trained/evaluated on actual data.

Could you please check if you are facing the same issue after training the model? Thanks!
